canonical/vanilla-framework
# Refactor theme colour variables to CSS custom properties

## Done

This PR introduces CSS custom properties (CSS variables) for theme colours to simplify theme implementation in components. This also introduces theme inheritance (so the theme set on parent element/component will be used by child components, unless overridden).

This PR focuses on core variables and some basic component updates. Updates of other components will follow up in individual PRs.

From code point of view this is refactoring, at this point it doesn't introduce new class names and should not change the visual appearance of components or introduce any regressions.

Fixes [WD-7555](https://warthogs.atlassian.net/browse/WD-7555)

## QA

- Open [demo](https://vanilla-framework-4921.demos.haus/)
- Review updated examples (themes should work as before without regressions)
  - applying `is-dark` or `is-light` class to given component or parent element should change the theme
  - HR light: https://vanilla-framework-4921.demos.haus/docs/examples/base/hr
  - HR dark: https://vanilla-framework-4921.demos.haus/docs/examples/base/hr-dark
  - Contextual menu light: https://vanilla-framework-4921.demos.haus/docs/examples/patterns/contextual-menu/default
  - Contextual menu dark: https://vanilla-framework-4921.demos.haus/docs/examples/patterns/contextual-menu/dark
  - Divider light: https://vanilla-framework-4921.demos.haus/docs/examples/patterns/divider/default
  - Divider dark: https://vanilla-framework-4921.demos.haus/docs/examples/patterns/divider/dark
- Review updated documentation:
  - https://vanilla-framework-4921.demos.haus/docs/settings/color-settings

### Check if PR is ready for release

If this PR contains Vanilla SCSS code changes, it should contain the following changes to make sure it's ready for the release:

  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
  - if CSS class names are not changed it can be bugfix relesase (x.x.**X**)
  - if CSS class names are changed/added/removed it should be minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical/vanilla-framework/wiki/Release-process#pre-release-tasks)



[WD-7555]: https://warthogs.atlassian.net/browse/WD-7555?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
